tests/components/backup/test_manager.py

```diff
@@ -1409,7 +1409,21 @@ async def test_initiate_backup_file_error(
     """Test file error during generate backup."""
     agent_ids = ["test.remote"]
     local_agent = local_backup_platform.CoreLocalBackupAgent(hass)
-    remote_agent = BackupAgentTest("remote", backups=[])
+
+    async def upload_backup(
+        *,
+        open_stream: Callable[[], Coroutine[Any, Any, AsyncIterator[bytes]]],
+        backup: AgentBackup,
+        **kwargs: Any,
+    ) -> None:
+        """Upload a backup."""
+        backup_stream = await open_stream()
+        backup_data = bytearray()
+        async for chunk in backup_stream:
+            backup_data += chunk
```

Comment:
This is used in three tests, we could add it to the mock returned by `mock_backup_agent`